src/command_modules/azure-cli-rdbms/azure/cli/command_modules/rdbms/custom.py

```diff
@@ -37,33 +43,56 @@ def _server_restore(cmd, client, resource_group_name, server_name, parameters, *
     except Exception as e:
         raise ValueError('Unable to get source server: {}.'.format(str(e)))
 
+def _restore_mysql(cmd, client, resource_group_name, server_name, parameters, **kwargs):
+    _server_restore(cmd, client, resource_group_name, server_name, parameters, **kwargs)
     return client.create_or_update(resource_group_name, server_name, parameters)
 
+def _restore_postgresql(cmd, client, resource_group_name, server_name, parameters, **kwargs):
+    _server_restore(cmd, client, resource_group_name, server_name, parameters, **kwargs)
+    return client.create(resource_group_name, server_name, parameters)
 
 def _server_update_custom_func(instance,
                                capacity=None,
+                               storage_mb=None,
+                               backup_retention_days=None,
                                administrator_login_password=None,
                                ssl_enforcement=None,
                                tags=None):
+    from azure.mgmt.rdbms.mysql.models import StorageProfile
     from importlib import import_module
     server_module_path = instance.__module__
     module = import_module(server_module_path.replace('server', 'server_update_parameters'))
     ServerUpdateParameters = getattr(module, 'ServerUpdateParameters')
 
     if capacity is not None:
+        instance.sku.name = _get_sku_name(instance.sku.tier, instance.sku.family, capacity)
         instance.sku.capacity = capacity
     else:
         instance.sku = None
 
+    if storage_mb is not None:
+        instance.storage_profile.storage_mb = storage_mb
+
+    if backup_retention_days is not None:
+        instance.storage_profile.backup_retention_days = backup_retention_days
+
     params = ServerUpdateParameters(sku=instance.sku,
-                                    storage_mb=None,
+                                    storage_profile=instance.storage_profile,
                                     administrator_login_password=administrator_login_password,
                                     version=None,
                                     ssl_enforcement=ssl_enforcement,
                                     tags=tags)
 
     return params
 
+def _create_server_for_mysql(cmd, client, resource_group_name, server_name, parameters, **kwargs):
+     return client.create_or_update(resource_group_name, server_name, parameters)
```

Comment:
I prefer to update the swagger file to keep the operationId of PUT server the same for mysql and pg, then you don't have to define two function for pg & mysql. (make them both client.create())